app/apollo/schema/channel.js

```diff
@@ -25,6 +25,14 @@ const channelSchema = gql`
     created: Date
     location: String
   }
+  input CustomChannelDataInput {
+    key: String!
+    val: String
+  }
+  type CustomChannelData {
```

Comment:
Type CustomChannelDataInput is a duplicate of type CustomChannelData, if possible, one of them should be removed.

Reply:
From what I understand, I thought you need to separate out GraphQL input types from output types. Is there a different way to handle passing in more complex objects? I've been reading from [here](https://graphql.org/learn/schema/#input-types)

Reply:
i do not know, maybe you can inherit one from another...

Reply:
From looking at the [spec](http://spec.graphql.org/June2018/#sec-Input-Objects), it doesn't seem like there's a way for input types to inherit from output types/interfaces or vice-versa. There's ways to extend Input types from other Input types, but that doesn't really help here. I think I could replace the inner `key` and `val` sections using template literals like this:
```
const customChannelType = `
    key: String!
    val: String
`;

const channelSchema = gql`
  // ...
  input CustomChannelDataInput {
    ${customChannelType}
  }
  type CustomChannelData {
    ${customChannelType}
  }
  // ...
`;
```

...but I'm not sure if that would be much cleaner. What do you think?

Reply:
no, let's keep the original. if the types were much more complex then it would make sense.

Reply:
@rmgraham may have a definitive answer on this. Check with him.